configuration/pih/petl/jobs/sql/extractions/mch_kellogg_data.sql

```diff
@@ -0,0 +1,333 @@
+set session group_concat_max_len = 100000;
+
+select program_id into @mchProgram from program where uuid = '41a2715e-8a14-11e8-9a94-a6cf71072f73';
+select concept_name(concept_from_mapping('PIH','YES'),@locale) into @yes;
+select concept_name(concept_from_mapping('PIH','NO'),@locale) into @no;
+select name into @socioEncName from encounter_type where uuid = 'de844e58-11e1-11e8-b642-0ed5f89f718b';
+select program_workflow_id into @mchWorkflow from program_workflow where uuid = '41a277d0-8a14-11e8-9a94-a6cf71072f73';
+set @past_med_finding = concept_from_mapping('PIH','10140');
+select encounter_type_id into @ob_gyn_enc_id from encounter_type where uuid = 'd83e98fd-dc7b-420f-aa3f-36f648b4483d';
+
+drop temporary table if exists temp_j9;
+
+create temporary table temp_j9
+(
+patient_id int,
+patient_program_id int,
+patient_age int,
+last_socio_encounter_id int,
+education_level varchar(255),
+able_read_write  boolean,
+date_enrolled datetime,
+date_completed datetime,
+program_state varchar(255),
+expected_delivery_date datetime,
+highest_birth_number_obs_group int,
+highest_birth_number int,
+prior_birth_delivery_type varchar(255), 
+prior_birth_neonatal_status varchar(255),
+history_pre_eclampsia boolean,
+history_eclampsia boolean,
+history_post_partum_hemorrhage boolean,
+history_gender_based_violence boolean, 
+history_type_1_diabetes boolean,
+history_type_2_diabetes boolean,
+history_gestational_diabetes boolean,
+number_anc_visit int, 
+number_obGyn_visits int,
+number_postpartum_visits int,
+number_family_planning_visits int,
+marital_status varchar(255), 
+employment_status varchar(255),
+religion varchar(255),
+family_support boolean,  
+partner_support_anc boolean,
+number_older_children int, 
+number_household_members int,
+address_department varchar(255),
+address_commune varchar(255),
+address_section_communale varchar(255),
+address_locality varchar(255),
+address_street_landmark varchar(255),
+access_transport boolean,
+mode_transport varchar(255), 
+traditional_healer varchar(255),
+prenatal_teas varchar(255) 
+);
+
+-- insert one row for every patient enrollment row 
+insert into temp_j9 (patient_id,patient_program_id,date_enrolled,date_completed)
+select patient_id,patient_program_id,date_enrolled,date_completed 
+from patient_program pp 
+where program_id = @mchProgram
+and voided = 0
+;
+
+-- encounter_id of latest encounter_id (used in calculations below) 
+update temp_j9
+set last_socio_encounter_id = latestEnc(patient_id, @socioEncName, null);
+
+update temp_j9 t
+set program_state = currentProgramState(t.patient_program_id, @mchWorkflow, @locale);
+
+
+-- patient age
+update temp_j9
+set patient_age = current_age_in_years(patient_id);
+
+-- columns from socioeconomic form
+-- education level
+update temp_j9
+set education_level = latest_obs_value_coded(patient_id, 'CIEL','1712',@locale);
+
+-- able to read or write
+update temp_j9
+set able_read_write = 
+CASE latest_obs_value_coded(patient_id, 'CIEL','166855',@locale)
```

Comment:
This looks like the kind of thing that should be wrapped in a function and reused

Reply:
Yeah I know...don't know why I was resisting it.  I guess it would be a function that transforms a yes/no to boolean that I wrap around the existing function.